floe/restapi.py

```diff
@@ -81,33 +83,33 @@ def set(self, key, value):
         resp = self.session.put("%s/%s" % (self._baseurl, key),
                                 data=value,
                                 headers={
-                                    'content-type': 'binary/octet-stream'}
+                                    'content-type': 'binary/octet-stream'},
+                                timeout=FLOE_REST_TIMEOUT
                                 )
         self.raise_exception_from_response(resp)
 
     def delete(self, key):
         key = sanitize_key(key)
-        resp = self.session.delete("%s/%s" % (self._baseurl, key))
+        resp = self.session.delete("%s/%s" % (self._baseurl, key), timeout=FLOE_REST_TIMEOUT)
         self.raise_exception_from_response(resp)
 
     def get_multi(self, keys):
-        responses = {}
         keys = [sanitize_key(key) for key in keys]
 
         def _get(key):
-            responses[key] = self.get(key)
+            return key, self.get(key)
 
-        self.pool.map(_get, keys)
+        responses = list(self.pool.map(_get, keys, timeout=FLOE_TASK_TIMEOUT))
 
-        return {k: v for k, v in responses.items() if v is not None}
+        return {k: v for k, v in responses if v is not None}
 
     def set_multi(self, mapping):
         mapping = {sanitize_key(key): value for key, value in mapping.items()}
 
         def _set(row):
             self.set(*row)
 
-        self.pool.map(_set, mapping.items())
+        list(self.pool.map(_set, mapping.items(), timeout=FLOE_TASK_TIMEOUT))
```

Comment:
Why `list()` here?  Does `map` return a lazy iterator that we need to force into loading?